control-plane/roles/isolated-clusters/tasks/ingress.yaml

```diff
@@ -0,0 +1,82 @@
+- name: Get kubeconfig
+  copy:
+    src: "files/kubeconfigs/kubeconfig.{{ cluster }}"
+    dest: "/tmp/kubeconfig.{{ cluster }}"
+
+- name: Create namespace for ingress-controller
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "{{ isolated_clusters_ingress_ingress_controller_namespace }}"
+    apply: yes
+  environment:
+    K8S_AUTH_KUBECONFIG: "/tmp/kubeconfig.{{ cluster }}"
+
+- name: Deploy ingress-controller
+  include_role:
+    name: ansible-common/roles/helm-chart
+    apply:
+      environment:
+        KUBECONFIG: "/tmp/kubeconfig.{{ cluster }}"
+  vars:
+    helm_repo: https://kubernetes.github.io/ingress-nginx
+    helm_chart_version: "{{ isolated_clusters_ingress_ingress_controller_chart_version }}"
+    helm_chart: ingress-nginx
+    helm_release_name: ingress-nginx
+    helm_target_namespace: "{{ isolated_clusters_ingress_ingress_controller_namespace }}"
+    helm_value_file_template: ingress/values.yaml
+
+- name: Deploy Cert Manager for Registry
+  include_role:
+    name: cert-manager
+    apply:
+      tags: cert-manager
+      environment:
+        KUBECONFIG: "/tmp/kubeconfig.{{ cluster }}"
+      vars:
+        cert_manager_version: "{{ isolated_clusters_ingress_cert_manager_version }}"
+        cert_manager_cluster_issuer: "{{ isolated_clusters_ingress_cert_manager_cluster_issuer }}"
+        cert_manager_lets_encrypt_expiry_mail_address: "{{ isolated_clusters_ingress_cert_manager_lets_encrypt_expiry_mail_address }}"
+        cert_manager_psp_enabled: false
+
+- name: Create dns secret for gcp cluster issuer
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: clouddns-serviceaccount-secret
+        namespace: cert-manager
+      type: Opaque
+      data: "{{ gardener_dns_credentials }}"
```

Comment:
These need to become role-prefixed variables, we should not pick up any undocumented variables.